test/unit/services/test_system.py

```diff
@@ -2,284 +2,374 @@
-    system_service,
-):
-    """
-    Test updating a System
+class GetBreadcrumbsDSL(SystemServiceDSL):
+    """Base class for get_breadcrumbs tests"""
 
-    Verify that the 'update' method properly handles the update of a System
-    """
-    system = SystemOut(
-        id=str(ObjectId()),
-        **{**SYSTEM_A_INFO_FULL, "created_time": SYSTEM_A_INFO_FULL["created_time"] - timedelta(days=5)},
-    )
+    _expected_breadcrumbs: MagicMock
+    _obtained_breadcrumbs: MagicMock
+    _obtained_system_id: str
 
-    # Mock `get` to return a System
-    test_helpers.mock_get(
-        system_repository_mock,
-        SystemOut(
-            id=system.id,
-            **{
-                **SYSTEM_A_INFO_FULL,
-                "name": "Different name",
-                "code": "different-name",
-                "created_time": system.created_time,
-                "modified_time": system.created_time,
-            },
-        ),
-    )
-    # Mock 'update' to return the updated System
-    test_helpers.mock_update(system_repository_mock, system)
-
-    updated_system = system_service.update(system.id, SystemPatchSchema(name=system.name))
-
-    system_repository_mock.update.assert_called_once_with(
-        system.id,
-        SystemIn(
-            **{
-                **SYSTEM_A_INFO_FULL,
-                "created_time": system.created_time,
-            }
-        ),
-    )
-    assert updated_system == system
-
-
-def test_update_with_non_existent_id(test_helpers, system_repository_mock, system_service):
-    """
-    Test updating a System with a non-existent ID
-
-    Verify that the 'update' method properly handles the update of a System with a non-existent ID
-    """
-    # Mock `get` to not return a System
-    test_helpers.mock_get(system_repository_mock, None)
-
-    system_id = str(ObjectId())
-    with pytest.raises(MissingRecordError) as exc:
-        system_service.update(system_id, MagicMock())
-    system_repository_mock.update.assert_not_called()
-    assert str(exc.value) == f"No System found with ID: {system_id}"
-
-
-def test_delete(system_repository_mock, system_service):
-    """
-    Test deleting a System
-
-    Verify that the `delete` method properly handles the deletion of a System by ID
-    """
-    system_id = MagicMock()
-
-    system_service.delete(system_id)
-
-    system_repository_mock.delete.assert_called_once_with(system_id)
+    def mock_get_breadcrumbs(self):
+        """Mocks repo methods appropriately to test the 'get_breadcrumbs' service method"""
+
+        # Simply a return currently, so no need to use actual data
+        self._expected_breadcrumbs = MagicMock()
+        self.test_helpers.mock_get_breadcrumbs(self.mock_system_repository, self._expected_breadcrumbs)
+
+    def call_get_breadcrumbs(self, system_id: str):
+        """Calls the SystemService `get` method"""
+
+        self._obtained_system_id = system_id
+        self._obtained_breadcrumbs = self.system_service.get_breadcrumbs(system_id)
```

Comment:
It's complaining about the type - `Expected type 'MagicMock', got 'BreadcrumbsGetSchema' instead`